specs/validator/beacon_node_oapi.yaml

```diff
@@ -0,0 +1,613 @@
+        current_version:
+          type: string
+          format: byte
+          pattern: "^0x[a-fA-F0-9]{8}$"
+          description: "Current fork version."
+        epoch:
+          type: integer
+          format: uint64
+          description: "Fork epoch number."
+    IndexedAttestation:
+      type: object
+      description: "The [`IndexedAttestation`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#indexedattestation) object from the Eth2.0 spec."
+      properties:
+        custody_bit_0_indicies:
+          type: array
+          description: "Validator indicies for 0 bits."
+          items:
+            type: integer
+            format: uint64
+        custody_bit_1_indicies:
+          type: array
+          description: "Validator indicies for 1 bits."
+          items:
+            type: integer
+            format: uint64
+        data:
+          $ref: '#/components/schemas/AttestationData'
+    AttestationData:
+      type: object
+      description: "The [`AttestationData`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#attestationdata) object from the Eth2.0 spec."
+      properties:
+        beacon_block_root:
+          type: string
+          format: byte
+          pattern: "^0x[a-fA-F0-9]{64}$"
+          description: "LMD GHOST vote."
+        source_epoch:
+          type: integer
+          format: uint64
+          description: "Source epoch from FFG vote."
+        source_root:
+          type: string
+          format: byte
+          pattern: "^0x[a-fA-F0-9]{64}$"
+          description: "Source root from FFG vote."
+        target_epoch:
+          type: integer
+          format: uint64
+          description: "Target epoch from FFG vote."
+        target_root:
+          type: string
+          format: byte
+          pattern: "^0x[a-fA-F0-9]{64}$"
+          description: "Target root from FFG vote."
+        crosslink:
+          title: CrossLink
+          type: object
+          description: "The [`Crosslink`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#crosslink) object from the Eth2.0 spec."
+          properties:
+            shard:
+              type: integer
+              format: uint64
+              description: "The shard number."
+            epoch:
+              type: integer
+              format: uint64
+              description: "The epoch number."
+            parent_root:
+              type: string
+              format: byte
+              pattern: "^0x[a-fA-F0-9]{64}$"
+              description: "Root of the previous crosslink."
+            data_root:
+              type: string
+              format: byte
+              pattern: "^0x[a-fA-F0-9]{64}$"
+              description: "Root of the crosslinked shard data since the previous crosslink."
+
+  responses:
+    Success:
+      description: "Request successful."
+    InvalidRequest:
+      description: "Invalid request syntax."
```

Comment:
Maybe define a new code `404` for the "Resource not found".

Reply:
I agree that an error `404` should be returned if a non-existent API endpoint is requested, however I don't think that a '404 - NotFound' error would be an appropriate return code for any of the actual endpoints (since you aren't requesting data which may or may not exist).
Since the actual error code is defined on the `path` (endpoint) definitions, it didn't seem appropriate to list a `NotFound` in components without referring to it somewhere in the endpoint definition. I agree with you however, that it would be nice to have it listed with all the other responses so that implementers know it is actually a valid response.